### PR TITLE
Scancode no exceptions

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

---
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/1459)

### CODE REVIEW

File: scanner/src/main/kotlin/scanners/ScanCode.kt [hunk 1]
```diff
@@ -488,7 +488,11 @@ class ScanCode(name: String, config: ScannerConfiguration) : LocalScanner(name,
         result["files"].forEach { file ->
             val path = file["path"].asText()
 
-            val licenses = file["licenses"]?.toList().orEmpty()
+            val licenses = file["licenses"]?.filter {
+                // TODO: Remove this again once we properly handle license expressions throught the code base and rules.
```
Comment (fviernau) [inline scanner/src/main/kotlin/scanners/ScanCode.kt]: throughout ?

File: scanner/src/main/kotlin/scanners/ScanCode.kt [hunk 2]
```diff
@@ -410,7 +410,7 @@ class ScanCode(name: String, config: ScannerConfiguration) : LocalScanner(name,
      */
     internal fun getClosestCopyrightStatements(
         path: String,
-        copyrights: JsonNode,
+        copyrights: List<JsonNode>,
```
Comment (fviernau) [inline scanner/src/main/kotlin/scanners/ScanCode.kt]: Please use 'Collection' as type to not specialize this without need.

File: scanner/src/main/kotlin/scanners/ScanCode.kt [hunk 3]
```diff
@@ -433,8 +433,8 @@ class ScanCode(name: String, config: ScannerConfiguration) : LocalScanner(name,
      */
     private fun associateFileFindings(
         path: String,
-        licenses: JsonNode,
-        copyrights: JsonNode,
+        licenses: List<JsonNode>,
```
Comment (fviernau) [inline scanner/src/main/kotlin/scanners/ScanCode.kt]: Please use Collection here as well.

File: scanner/src/main/kotlin/scanners/ScanCode.kt (hunk 1 above)
Comment (sschuberth) [reply scanner/src/main/kotlin/scanners/ScanCode.kt]: Thanks, done.

File: scanner/src/main/kotlin/scanners/ScanCode.kt (hunk 2 above)
Comment (sschuberth) [reply scanner/src/main/kotlin/scanners/ScanCode.kt]: Done.

File: scanner/src/main/kotlin/scanners/ScanCode.kt (hunk 3 above)
Comment (sschuberth) [reply scanner/src/main/kotlin/scanners/ScanCode.kt]: Done.

**Pull request merged**